sdk/eventhub/event-hubs/src/linkEntity.ts

```diff
@@ -126,9 +135,16 @@ export class LinkEntity {
       this.name,
       this.address
     );
-    await defaultLock.acquire(this._context.cbsSession.cbsLock, () => {
-      return this._context.cbsSession.init();
-    });
+    await defaultCancellableLock.acquire(
+      this._context.cbsSession.cbsLock,
+      () => {
+        return this._context.cbsSession.init({ abortSignal });
```

Comment:
Should we do a check if the cbsSesion is already open before wasting an attempt to get a lock?

Reply:
Sure. I'll have to expose a new method on the CbsClient in core-amqp so I can tell that it's been opened, but I can do ahead and do that in this PR.

Reply:
Method added!